astropy/utils/masked/tests/test_function_helpers.py

```diff
@@ -1453,6 +1453,185 @@ def test_nanpercentile(self):
         self.check(np.nanpercentile, q=50)
 
 
+class TestArraySetOps:
+    """Tests based on those from numpy.ma.tests.test_extras.
+
+    Adjusted to take into account that comparing masked values should
+    result in masked equality.
+
+    """
+
+    @classmethod
+    def setup_class(self):
+        # Setup for unique.
+        self.data = Masked([1, 1, 1, 2, 2, 3], mask=[0, 0, 1, 0, 1, 0])
+        self.values = Masked([1, 2, 3, 1, 2], mask=[0, 0, 0, 1, 1])
+        self.indices = np.array([0, 3, 5, 2, 4])
+        self.inverse_indices = np.array([0, 0, 3, 1, 4, 2])
+        self.counts = np.array([2, 1, 1, 1, 1])
+
+    @pytest.mark.parametrize("dtype", [int, float, object])
+    def test_unique(self, dtype):
+        test = np.unique(
+            self.data.astype(dtype), return_index=True, return_inverse=True
+        )
+        assert_masked_equal(test[0], self.values.astype(dtype))
+        assert_array_equal(test[1], self.indices)
+        assert_array_equal(test[2], self.inverse_indices)
+        # All masked
+        data2 = Masked([2, 1, 3], mask=True)
+        test2 = np.unique(data2.astype(dtype), return_index=True, return_inverse=True)
+        values2 = Masked([1, 2, 3], mask=True)
+        assert_masked_equal(test2[0], values2.astype(dtype))
+        assert_array_equal(test2[1], [1, 0, 2])
+        assert_array_equal(test2[2], [1, 0, 2])
+
+    if not NUMPY_LT_2_0:
```

Comment:
also nit, but I have a slight preference for using `@pytest.skipif(NUMPY_LT_2_0)` here for two reasons:
- it avoids a double negative
- removing the decorator later won't mess `git blame`, but changing the indentation will

Reply:
Good point. I worried this would not work with your function counting/consistency framework, but it does; nice!